test_runner/batch_others/test_branching.py

```diff
@@ -87,3 +88,35 @@ def run_pgbench(pg: Postgres):
     for pg in pgs:
         res = pg.safe_psql('SELECT count(*) from pgbench_accounts')
         assert res[0] == (100000 * scale, )
+
+
+# Test branching from an "unormalized" LSN.
+#
+# Context:
+# When doing basebackup for a newly created branch, pageserver generates
+# 'pg_control' file to bootstrap WAL segment by specifying the redo position
+# a "normalized" LSN based on the timeline's starting LSN:
+#
+# checkpoint.redo = normalize_lsn(self.lsn, pg_constants::WAL_SEGMENT_SIZE).0;
+#
+# This test checks if the pageserver is able to handle a "unormalized" starting LSN.
+#
+# Related: see discussion in https://github.com/neondatabase/neon/pull/2143#issuecomment-1209092186
+def test_branching_unnormalized_start_lsn(neon_simple_env: NeonEnv, pg_bin: PgBin):
+    XLOG_BLCKSZ = 8192
+
+    env = neon_simple_env
+
+    env.neon_cli.create_branch('b0')
+    pg0 = env.postgres.create_start('b0')
+
+    pg_bin.run_capture(['pgbench', '-i', pg0.connstr()])
+
+    curr_lsn = lsn_from_hex(pg0.safe_psql("SELECT pg_current_wal_flush_lsn()")[0][0])
+    start_lsn = (curr_lsn - XLOG_BLCKSZ) // XLOG_BLCKSZ * XLOG_BLCKSZ
```

Comment:
can you add a comment here, what the math here does to 'start_lsn`? Is it `curr_lsn` rounded down to nearest XLOG_BLCKSZ boundary?